### PR TITLE
Add an image for ubuntu2204 with clang-built deps

### PR DESCRIPTION
**Pull request by paulgessinger**

### CODE REVIEW

File: ubuntu2204_clang/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DHEPMC3_BUILD_STATIC_LIBS=OFF \
+    -DHEPMC3_ENABLE_PYTHON=OFF \
+    -DHEPMC3_ENABLE_ROOTIO=OFF \
+    -DHEPMC3_ENABLE_SEARCH=OFF \
+  && cmake --build build -- install \
+  && rm -rf build src
+
+env CC=clang
+env CXX=clang++
```
Comment (andiwand) [inline ubuntu2204_clang/Dockerfile]: is there a reason not to use clang for `Geant4` and `HepMC3`?

**Review by andiwand — APPROVED** [review]
lgtm :+1:

File: ubuntu2204_clang/Dockerfile (hunk 1 above)
Comment (paulgessinger) [reply ubuntu2204_clang/Dockerfile]: oh, no just forgot to move it up. HepMC3 and Geant4 get clang anyway automatically through cmake.

**Comment by andiwand** [comment]
thinking about it we could actually pass the `ENV` by docker and spare the copy of ubuntu 2204. but that would break the architecture of the current CI yaml right?

**Comment by paulgessinger** [comment]
In principle yeah, but I'd honestly prefer if the image just didn't even have GCC installed in the first place, then you can't configure any builds wrong.

**Pull request merged**